crates/checkpoint-downloader/src/downloader.rs

```diff
@@ -362,18 +436,25 @@ impl ParallelCheckpointDownloaderInner {
         _config: &ParallelDownloaderConfig,
         _rng: &mut StdRng,
     ) -> CheckpointEntry {
-        let res = client.get_full_checkpoint(sequence_number).await;
-        let Ok(checkpoint) = res else {
-            let err = res.err();
-            handle_checkpoint_error(err.as_ref(), sequence_number);
-            let err = err
-                .map(|e| e.into())
-                .unwrap_or_else(|| anyhow!("Failed to download checkpoint"));
-            return CheckpointEntry::new(sequence_number, Err(err));
-        };
-        CheckpointEntry {
-            sequence_number,
-            result: Ok(checkpoint),
+        let res = tokio::time::timeout(
```

Comment:
suggeestion: remove this timeout. I think all the RPCs inside RetriableRpcClient::get_full_checkpoint already contains tokio::time::timoout.

Also, given this also includes fallback path, 1 second timeout feels too short.

Reply:
This timeout is only meant for test actually, to make the tests run quicker (since we know these calls will fail but we want them to fail quick in tests)